File: 0283-move-zeroes/0283-move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        j = 0 #i pura array traverse karega
        # j → batata hai next non-zero kahan rakhna hai
        #Jab non-zero mile:
        #usko aage shift karo (swap)
        #j++
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[j], nums[i] = nums[i], nums[j]
                j += 1
        

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {

        int index = 0;
        // Saare non-zero elements aage laao
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                nums[index] = nums[i];
                index++;
            }
        }

        // Bachi hui positions mein 0 bhar do
        for (int i = index; i < nums.length; i++) {
            nums[i] = 0;
        }
    }
}